core/src/wsv.rs

```diff
@@ -307,13 +307,14 @@ impl<W: WorldTrait> WorldStateView<W> {
 
     /// Send [`Event`]s to known subscribers.
     fn produce_event(&self, event: impl Into<Event>) {
-        let events_sender = if let Some(sender) = &self.events_sender {
-            sender
-        } else {
-            return warn!("wsv does not equip an events sender");
-        };
-
-        drop(events_sender.send(event.into()))
+        self.events_sender.as_ref().map_or_else(
```

Comment:
So how exactly does this resolve the issue?

Reply:
There was no issue to begin with. The initial code was not understandable (to me) so I thought there must be some quirk with the tracing crate

Reply:
Personally, I prefer [early returns](https://doc.rust-lang.org/rust-by-example/error/result/early_returns.html) to avoid nesting

Reply:
> Personally, I prefer [early returns](https://doc.rust-lang.org/rust-by-example/error/result/early_returns.html) to avoid nesting

I think we can all agree in favor of early returns

Reply:
> Personally, I prefer [early returns](https://doc.rust-lang.org/rust-by-example/error/result/early_returns.html) to avoid nesting

I agree that early returns are great to avoid nesting, but in this case we have the same level of nesting.
In my opinion, the problem with initial code was that [return](https://doc.rust-lang.org/std/keyword.return.html) is usually used to return some value from function, but in this case we combine it with `warn` macro that is not supposed to return any value.